source/validate.cpp

```diff
@@ -234,6 +234,81 @@ UNUSED(void PrintDotGraph(ValidationState_t& _, libspirv::Function func)) {
   }
 }
 
+// Returns the major version number from |version|.
+uint32_t parseMajorVersion(uint32_t version) {
+  // 2.3 Table 1: Version bytes are 0 | major | minor | 0.
+  return version >> 16;
+}
+
+// Returns the minor version number from |version|.
+uint32_t parseMinorVersion(uint32_t version) {
+  // 2.3 Table 1: Version bytes are 0 | major | minor | 0.
+  return (version & 0xff00) >> 8;
+}
+
+// Returns true if SPIR-V version |version| supports the target environment
+// |env|.
+bool validateVersion(uint32_t version, const spv_target_env& env) {
+  uint32_t major = parseMajorVersion(version);
+  uint32_t minor = parseMinorVersion(version);
+  switch (env) {
+    case SPV_ENV_VULKAN_1_0:
```

Comment:
This function should use spvVersionForTargetEnv from spirv_constant.h
Then you won't need any specific cases, and that will reduce maintenance.

Reply:
(And feel free to move that function from spirv_constant.h to spirv_target_env.h which seems like a more discoverable place for it.